File: reset_password.php
<?php
require_once 'config.php';

// Script pour lister les utilisateurs et réinitialiser un mot de passe
echo "=== Gestion des utilisateurs ===\n\n";

$db = getDB();

// Lister tous les utilisateurs
echo "Liste des utilisateurs :\n";
$stmt = $db->query("SELECT id, username FROM users");
$users = $stmt->fetchAll(PDO::FETCH_ASSOC);

foreach ($users as $user) {
    echo "  - ID: {$user['id']}, Username: {$user['username']}\n";
}

echo "\n";

// Fonction pour réinitialiser le mot de passe
if ($argc > 2) {
    $username = $argv[1];
    $newPassword = $argv[2];

    $hashedPassword = password_hash($newPassword, PASSWORD_DEFAULT);

    $stmt = $db->prepare("UPDATE users SET password = ? WHERE username = ?");
    $stmt->execute([$hashedPassword, $username]);

    if ($stmt->rowCount() > 0) {
        echo "✅ Mot de passe mis à jour pour l'utilisateur : $username\n";
        echo "   Nouveau mot de passe : $newPassword\n";
    } else {
        echo "❌ Utilisateur non trouvé : $username\n";
    }
} else {
    echo "Pour réinitialiser un mot de passe, utilisez :\n";
    echo "php reset_password.php <username> <nouveau_mot_de_passe>\n";
}

File: reset_password.py
#!/usr/bin/env python3
import sqlite3
import sys
import hashlib

# Connexion à la base de données
db_path = '/home/cryborg/Projects/arcade-franck/tools/accounts/data/accounts.db'
conn = sqlite3.connect(db_path)
cursor = conn.cursor()

print("=== Gestion des utilisateurs ===\n")

# Lister tous les utilisateurs
print("Liste des utilisateurs :")
cursor.execute("SELECT id, username FROM users")
users = cursor.fetchall()

for user in users:
    print(f"  - ID: {user[0]}, Username: {user[1]}")

print("\n")

# Fonction pour réinitialiser le mot de passe (avec bcrypt en PHP)
if len(sys.argv) > 2:
    username = sys.argv[1]
    new_password = sys.argv[2]

    print(f"⚠️  ATTENTION : Ce script ne peut pas créer de hash bcrypt PHP compatible.")
    print(f"Pour réinitialiser le mot de passe de '{username}', vous devez :")
    print(f"1. Soit utiliser PHP directement sur le serveur de production")
    print(f"2. Soit créer un nouveau compte depuis https://devplayground.ovh/tools/accounts/register.php")
    print(f"3. Soit me demander de créer un script PHP à uploader sur le serveur")
else:
    print("Pour réinitialiser un mot de passe, utilisez :")
    print("python3 reset_password.py <username> <nouveau_mot_de_passe>")

conn.close()
